### PR TITLE
feat(frontend): Add fridge component

### PR DESCRIPTION
**Pull request by JJTomines**
- Added Fridge to nav menu
- Add a list that displays ingredients in the fridge, a placeholder message if empty
- Can click button on an ingredient to add or remove
- Can use the search bar to filter items that will appear with added items appearing first in the list
- Add fridge service for getting ingredients

### CODE REVIEW

File: client/src/app/fridge/fridge.component.scss [hunk 1]
```diff
@@ -0,0 +1,53 @@
+.ingredient {    
+    border: 2px;
+    border-radius: 10px;
+    margin-top: 2.5%;
+    margin-bottom: 2.5%;
+    padding: 20px 10px 20px 10px;
+   }
+
+.availableIngredient {
```
Comment (zaidi-01) [inline client/src/app/fridge/fridge.component.scss:9]: nit: follow HTML CSS naming convention.

File: client/src/app/fridge/fridge.component.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+@Component({
+  standalone: true,
+  selector: 'app-fridge-component',
+  templateUrl: './fridge.component.html',
+  styleUrls:['./fridge.component.scss'],
+  imports: [
+    CommonModule,
+    FormsModule,
+    NgIf,
+    NgFor,
```
Comment (zaidi-01) [inline client/src/app/fridge/fridge.component.ts:22]: Import these in the app module instead.

File: client/src/app/ingredient.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+//Test list of data, will be replaced with DB
+
+export interface Ingredient {
```
Comment (zaidi-01) [inline client/src/app/ingredient.ts:3]: I think we should create a `FridgeService` at this point. Store the service and related files under `client/src/services/fridge`. @StevieShibly8 @NilinReza What do you guys think?

File: client/src/app/mock-ingredients.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { Ingredient } from './ingredient';
+
+export const INGREDIENTS: Ingredient[] = [
```
Comment (zaidi-01) [inline client/src/app/mock-ingredients.ts:3]: This mock data will be provided by the `FridgeService` for the time being until backend API is ready.

File: client/src/app/fridge/fridge.component.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    UpperCasePipe,
+  ],
+})
+export class FridgeComponent implements OnInit{
+
+  addedIngredients: Set<string> = this.getLocallyStoredIngredients();
+  //TODO: Replace this list of ingredients with one found in ingredients table once available
+  availableIngredients = [...INGREDIENTS].map(ingredient => ingredient.name).sort();
```
Comment (zaidi-01) [inline client/src/app/fridge/fridge.component.ts:30]: Replace with a `FridgeService` call, that way we only have to refactor the service in the future.

File: client/src/app/ingredient.ts (hunk 3 above)
Comment (JJTomines) [reply client/src/app/ingredient.ts]: FridgeService implemented for getting available ingredients and local storage ingredients

File: client/src/app/fridge/fridge.component.ts (hunk 2 above)
Comment (JJTomines) [reply client/src/app/fridge/fridge.component.ts]: moved to app.module.ts

File: client/src/app/fridge/fridge.component.scss (hunk 1 above)
Comment (JJTomines) [reply client/src/app/fridge/fridge.component.scss]: renamed .availableIngredient to .available-ingredient, that's more inline with the naming convention now right?

**Review by zaidi-01 — APPROVED** [review]

**Review by StevieShibly8 — APPROVED** [review]

**Review by zaidi-01 — APPROVED** [review]

**Review by StevieShibly8 — APPROVED** [review]

**Pull request merged**